kedro/config/omegaconf_config.py

```diff
@@ -275,7 +277,13 @@ def load_and_merge_dir_config(
             return {}
         if len(aggregate_config) == 1:
             return list(aggregate_config)[0]
-        return dict(OmegaConf.merge(*aggregate_config))
+
```

Comment:
I think the flow here is a little unclear (too many different `return` cases) and, more importantly, unless I'm missing something there's actually a bug: if you have a single element in `aggregate_config` (i.e. one parameters file) then `runtime_params` don't get merged in.

This would be solved if we removed lines 276-279. I know @merelcht said there was some reason to pick out the `len == 1` case before but I don't see what this could be now that we always want to do `to_container(resolve=True)`.

Reply:
I would be keen on removing it, unless we have a specific failing case. I try remove line 278-279 and all test case passed.

Should we separate this into another issue? Removing Line 276-277 will cause a few tests fail, I suspect it's also link to https://github.com/kedro-org/kedro/pull/2556 and some refactoring may be needed.

Reply:
I'm happy for those lines to be removed if everything is still working fine. I don't remember why I added it, so really should have left a comment 😅

Reply:
I think we should remove line 278-279 here since as I understand it at the moment they are not just unnecessary but following this change will actually be causing a bug (would be good if you could check this): if you just have a single parameters.yml file, the `runtime_params` don't get merged in.

Lines 276-277 don't actually cause a bug I think, so if removing them makes some tests fail and it seems related to #2556 then let's leave those here for now and try to remove them in that PR.